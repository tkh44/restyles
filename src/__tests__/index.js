/* eslint-disable jsx-quotes */
/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer';
import {matcher, serializer} from 'jest-glamor-react'
import Style, {css} from '../index'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

describe('restyles', () => {
  test('css function', () => {
    let element = css({ color: 'blue' })
    const tree = renderer.create(element).toJSON()
    expect(tree).toMatchSnapshotWithGlamor()
  })

  test('renders elements with styles', () => {
    const props = {
      online: true,
      theme: {radius: 5}
    }

    const tree = renderer
      .create(
        <Style>
          {css({
            '& .profile': {
              color: 'green',
              fontSize: 48,
              '&:hover': {
                color: 'blue'
              },
              '& > span': {
                fontSize: 64,
                marginLeft: 'auto'
              }
            },
            '& .activity': {
              color: props.online ? 'green' : 'gray',
              display: 'flex',
              '& .moment': {
                color: 'rebeccapurple',
                fontSize: 18
              }
            },
            '@media(min-width: 400px)': {
              '& .moment': {
                height: 200
              }
            }
          })}
          <div className="profile">
            <h1>I'm blue sometimes <span>ðŸ‘¾</span></h1>
          </div>
          <div className="activity">
            <div className="moment">Buy Beer</div>
            <div className="moment">Add Server</div>
            <div className="moment">Log In</div>
          </div>
        </Style>
      )
      .toJSON()

    expect(tree).toMatchSnapshotWithGlamor()
  })
})
