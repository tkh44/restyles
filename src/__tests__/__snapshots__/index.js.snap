// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restyles css function 1`] = `<noscript />`;

exports[`restyles renders elements with styles 1`] = `
Array [
  <noscript />,
  .css-vwt4h7 .profile,
[data-css-vwt4h7] .profile {
  color: green;
  font-size: 48px;
}

.css-vwt4h7 .profile:hover,
[data-css-vwt4h7] .profile:hover {
  color: blue;
}

.css-vwt4h7 .profile > span,
[data-css-vwt4h7] .profile > span {
  font-size: 64px;
  margin-left: auto;
}

.css-vwt4h7 .activity,
[data-css-vwt4h7] .activity {
  color: green;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
}

.css-vwt4h7 .activity .moment,
[data-css-vwt4h7] .activity .moment {
  color: rebeccapurple;
  font-size: 18px;
}

@media (min-width: 400px) {
  .css-vwt4h7 .moment,
  [data-css-vwt4h7] .moment {
    height: 200px;
  }
}

<div
    className="profile"
    data-css-vwt4h7=""
>
    <h1>
        I'm blue sometimes 
        <span>
            ðŸ‘¾
        </span>
    </h1>
</div>,
  .css-vwt4h7 .profile,
[data-css-vwt4h7] .profile {
  color: green;
  font-size: 48px;
}

.css-vwt4h7 .profile:hover,
[data-css-vwt4h7] .profile:hover {
  color: blue;
}

.css-vwt4h7 .profile > span,
[data-css-vwt4h7] .profile > span {
  font-size: 64px;
  margin-left: auto;
}

.css-vwt4h7 .activity,
[data-css-vwt4h7] .activity {
  color: green;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
}

.css-vwt4h7 .activity .moment,
[data-css-vwt4h7] .activity .moment {
  color: rebeccapurple;
  font-size: 18px;
}

@media (min-width: 400px) {
  .css-vwt4h7 .moment,
  [data-css-vwt4h7] .moment {
    height: 200px;
  }
}

<div
    className="activity"
    data-css-vwt4h7=""
>
    <div
        className="moment"
    >
        Buy Beer
    </div>
    <div
        className="moment"
    >
        Add Server
    </div>
    <div
        className="moment"
    >
        Log In
    </div>
</div>,
]
`;
